src/Compilers/CSharp/Test/Semantic/Semantics/LambdaTests.cs

```diff
@@ -7943,5 +7940,41 @@ public void ParamsArray_Symbol_MultipleParamsArrays()
             Assert.False(((SourceParameterSymbol)lambdas[1].Parameters[1]).IsParams);
             Assert.True(((SourceParameterSymbol)lambdas[1].Parameters[2]).IsParams);
         }
+
+        [Fact]
+        public void ParamsArray_NotLast()
+        {
+            var source = """
+                var lam = (params int[] xs, int y) => xs.Length + y;
+                """;
+            CreateCompilation(source).VerifyDiagnostics(
+                // (1,12): error CS0231: A params parameter must be the last parameter in a parameter list
+                // var lam = (params int[] xs, int y) => xs.Length + y;
+                Diagnostic(ErrorCode.ERR_ParamsLast, "params int[] xs").WithLocation(1, 12));
+        }
+
+        [Fact]
+        public void ParamsArray_Multiple()
```

Comment:
@cston I'm looking at this PR and thinking about the working group you're leading on expanding `params` support in the future to have more types. Feel like if that effort goes forward we'll just be back in this file in six months or so adding another round of tests, and then adding those tests for local funcs, normal funcs, etc ... Have we given any thought as to how we could centralize these tests so that we do less work on them?

Reply:
> I could create a separate test class with theories for all those cases ...

The `[Theory]` examples seem reasonable to me. Perhaps let's just start there and leave any additional refactoring until we know we need it. I think I'd prefer to keep the tests in LambdaTests.cs rather than creating a new test file though.

Reply:
> I'd prefer to keep the tests in LambdaTests.cs rather than creating a new test file though

Even though they are not testing just lambdas anymore, but also local functions and normal methods?

Reply:
> Even though they are not testing just lambdas anymore, but also local functions and normal methods?

Ah, I'd overlooked that, thanks.

In that case, perhaps let's wait until we have a need to refactor (if we want to reuse tests for additional `params` types for instance) before changing the tests.